File: python_75/twoSum.py
# Given an array of integers and a target, find 2 numbers that add up to target


class Solution(object):

  def twoSum(self, nums, target):
    i = 0
    diff_dict = {}
    for num in nums:
      diff = target - num
      if (diff in diff_dict):
        list = []
        list.append(nums[i])
        list.append(nums[diff_dict[diff]])
        return list
      else:
        diff_dict[num] = i
      i+= 1

p1 = Solution()
print(p1.twoSum([2, 3, 5, 6], 7))

File: leetcode/src/main/java/org/example/redo/TwoSum.java
package org.example.redo;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class TwoSum {

    public static void main(String[] args) {
        int[] nums = {2, 5, 7, 8};
        System.out.println(Arrays.toString(find(nums, 9)));
    }

    public static int[] find(int[] nums, int target) {
        Map<Integer, Integer> diffMap = new HashMap<>();
        int i=0;
        for (int num : nums) {
            int diff = target - num;
            if (diffMap.containsKey(diff)) {
                return new int[]{num, nums[diffMap.get(diff)]};
            } else {
                diffMap.put(num, i);
            }
            i++;
        }
        return null;
    }
}
